core/src/main/java/org/jdbi/v3/core/statement/ParsedParameters.java

```diff
@@ -78,4 +79,29 @@ public String toString() {
             + ", parameterNames=" + parameterNames
             + '}';
     }
+
+    /**
+     * A static factory of named {@link ParsedParameters} instances.
+     * List given in argument must contain parameter names from SQL.
+     * Names from a list must be stripped of a colon which stands
+     * before variable in unprocessed SQL.
+     *
+     * @param names the parameter names from SQL statement
+     * @return New {@link ParsedParameters} instance
+     */
+    public static ParsedParameters named(List<String> names) {
+        return new ParsedParameters(false, names);
```

Comment:
Since this method routes around the the usual builder interface, we should validate that `names` does not contain `POSITIONAL_PARAM`

Reply:
Done.

Reply:
Added `if` statement which throws `IllegalArgumentException` if `names.contains("?")`